python/graphstorm/model/utils.py

```diff
@@ -250,33 +289,103 @@ def save_embeddings(model_path, embeddings, local_rank, world_size):
         #     - owner' group can read, write;
         #     - others can read, write, and execute.
         os.chmod(model_path, 0o767)
+
     # make sure the model_path permission is changed before other process start to save
     th.distributed.barrier()
 
     assert local_rank < world_size
-    def get_data_range(num_embs):
-        # Get corresponding data range
-        start = local_rank * (num_embs // world_size)
-        end = (local_rank + 1) * (num_embs // world_size)
-        end = num_embs if local_rank + 1 == world_size else end
-        return start, end
+    backend = th.distributed.get_backend()
+    device = th.device('cpu') if backend == "gloo" else device
+    def exchange_node_id_mapping(node_id_mapping, num_embs):
```

Comment:
it's better to move the function out. it's not easy to read. it also makes the main function too long.

Reply:
Updated